src/components/spatial/utils.js

```diff
@@ -1,5 +1,215 @@
+ */
+function buildDefaultSelection(imageDims) {
+  const selection = [];
+  const globalSelection = getDefaultGlobalSelection(imageDims);
+  // First non-global dimension with some sort of selectable values
+  const firstNonGlobalDimension = imageDims.filter(
+    dim => !GLOBAL_SLIDER_DIMENSION_FIELDS.includes(dim.field) && dim.values,
+  )[0];
+  for (let i = 0; i < Math.min(4, firstNonGlobalDimension.values.length); i += 1) {
+    selection.push(
+      {
+        [firstNonGlobalDimension.field]: i,
+        ...globalSelection,
+      },
+    );
+  }
+  return selection;
+}
+
+/**
+ * Initialize the channel settings for an individual channel and selection.
+ * @param {object} loader A viv loader instance, for either Zarr or OME-TIFF.
+ * @param {object} selection The channel selection object.
+ * @param {number} i The index of this channel for the layer.
+ * @returns {object} The initialized channel with
+ * domain/slider settings.
+ */
+export async function initializeChannelForSelection(loader, selection, i) {
+  // Get stats because initial value is Min/Max for domainType.
+  const stats = await getChannelStats({ loader, loaderSelection: [selection] });
+
+  const domain = loader.isRgb ? [[0, 255], [0, 255], [0, 255]][i] : stats[0].domain;
+  const color = loader.isRgb ? [[255, 0, 0], [0, 255, 0], [0, 0, 255]][i] : VIEWER_PALETTE[i];
+  const slider = loader.isRgb ? [[0, 255], [0, 255], [0, 255]][i] : stats[0].autoSliders;
+
+  return {
+    selection,
+    domain,
+    color,
+    visible: true,
+    slider: slider || domain,
+  };
+}
+
+/**
+ * Initialize the channel selections for an individual layer.
+ * @param {object} loader A viv loader instance, for either Zarr or OME-TIFF.
+ * @returns {object[]} An array of selected channels with default
+ * domain/slider settings.
+ */
+export async function initializeLayerChannels(loader) {
+  const result = [];
+  const loaderDimensions = loader.dimensions;
+  // Add channel automatically as the first avaialable value for each dimension.
+  const defaultSelection = buildDefaultSelection(loaderDimensions);
+  // Get stats because initial value is Min/Max for domainType.
+  const stats = await getChannelStats({ loader, loaderSelection: defaultSelection });
+
+  const domains = loader.isRgb
+    ? [[0, 255], [0, 255], [0, 255]]
+    : stats.map(stat => stat.domain);
+  const colors = loader.isRgb
+    ? [[255, 0, 0], [0, 255, 0], [0, 0, 255]]
+    : null;
+  const sliders = loader.isRgb
+    ? [[0, 255], [0, 255], [0, 255]]
+    : stats.map(stat => stat.autoSliders);
+
+  defaultSelection.forEach((selection, i) => {
+    const domain = domains[i];
+    const slider = sliders[i];
+    const channel = {
+      selection,
+      domain,
+      color: colors ? colors[i] : VIEWER_PALETTE[i],
+      visible: true,
+      slider: slider || domain,
+    };
+    result.push(channel);
+  });
+  return result;
+}
+
+export async function initializeLayerChannelsIfMissing(layerDefsOrig, loaders) {
```

Comment:
Can we have some comments here (and where this is used) about why this is needed?  For example, the reference on 135 to `channels` might seem odd if you didn't know that the purpose of it is to filter out other, non-raster layers (if I have that right).

Reply:
Done. This function allows the user to pass an array of channel selections which may be missing the `.slider`, `.domain`, `.color`, or `.visible` properties (the function will auto-initialize these properties if they are missing for any of the selected channels).